src/app/adminLayout/students/student-details/student-details.component.ts

```diff
@@ -15,24 +17,23 @@ export class StudentDetailsComponent implements OnInit {
   @Output() stopDisplayEvent = new EventEmitter<boolean>();
   @Output() newStudentEvent = new EventEmitter<Student>();
   @Output() deleteOldStudentEvent = new EventEmitter<number>();
-  stateOptions: any[];
+  stateOptions: any[] = [];
   editName: boolean;
   editLastName: boolean;
-  editEmail:boolean;
+  editEmail: boolean;
   updateForm: FormGroup;
+  inputTextElements: InputTextElement[] = [];
```

Comment:
te nazwy, też bym zmienił na coś w stylu `inputStudentDetailsElements`, lub cokolwiek co się samo-dokumentuje, chyba że nie do tego to służy :D ale taka jest moja szybka interpretacja ;p

Reply:
Tutaj jeszcze poszukam, bo może istnieje jakaś predefiniowana klasa do wykorzystania w polach formularza, ogólnie chodziło o taki uniwersalny element przechowujący wszystkie rzeczy zmieniające się w formularzu (w sumie każdy, nie tylko student-detail, dlatego nazwa ogólna), żeby można było przez to przeiterować w htmlu.

Reply:
to może lepiej `inputFormElements` , albo cokolwiek z `Form`